Mathlib/Algebra/Group/Submonoid/Operations.lean

```diff
@@ -301,6 +301,14 @@ theorem comap_iInf {ι : Sort*} (f : F) (s : ι → Submonoid N) :
 theorem map_bot (f : F) : (⊥ : Submonoid M).map f = ⊥ :=
   (gc_map_comap f).l_bot
 
+@[to_additive (attr := simp)]
+theorem map_top_of_surjective (f : M →* N) (h : Function.Surjective f) :
+    Submonoid.map f ⊤ = ⊤ := by
+  rw [eq_top_iff]
+  intro x _
+  obtain ⟨y, hy⟩ := h x
+  exact ⟨y, trivial, hy⟩
```

Comment:
I think there's a shorter proof here of `rw [← MonoidHom.mrange_eq_map, MonoidHom.mrange_eq_top]`.

Reply:
Though I'd be tempted to just deprecate `Subgroup.map_top_of_surjective`, since really `MonoidHom.range_eq_top` is a much more useful lemma.

Reply:
I disagree about the deprecation because I think it's a useful simp lemma. Indeed, I ran across this because a proof for Group failed for Monoid due to it being missing.

`mrange_eq_map` is defined later in this file. I could move this lemma after it, but it would then not match it's position in the matching Subgroup file.

Reply:
I think that we should `simp`lify `map _ \top` to `mrange`/`range`.